### PR TITLE
Feature/v2 updates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR

**What change(s) does this PR introduce?** 

This PR includes the following updates:
- Removal of the `adwords` compatible models
- Removal of the `google_ads__ad_adapter` model
- Addition of the following new models:
  - `google_ads__account_report`
  - `google_ads__campaign_report`
  - `google_ads__ad_group_report`
  - `google_ads__keyword_report`
  - `google_ads__ad_report`
  - `google_ads__utm_report`
- Added testing for each end model to ensure granularity and accuracy of the modeled data.

**Did you update the CHANGELOG?** 

- [X] Yes

**Does this PR introduce a breaking change?**

- [X] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [X] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [X] No 

**How did you test the PR changes?** 

 

- [X] CircleCi  
- [X] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [X] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🤌 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Comment by fivetran-joemarkiewicz** [comment]
FYI the base branch is `feature/google-ads-api-default` as that branch will be merged prior to this one and I want to make sure I am not duplicating reviews from the changes made in PR #32

File: README.md
Comment (fivetran-sheringuyen) [inline README.md:70]: Can we also add in a little blurb here about doing some due diligence before including/interpreting additional metrics? I want to make sure we have it documented that the metrics that we've included have been vetted for aggregation, however we can't ensure that all additional passthrough metrics will make sense with aggregation

File: DECISIONLOG.md
Comment (fivetran-sheringuyen) [inline DECISIONLOG.md:7]: What are your thoughts on including the non-null adjustment to the UTM report here? I think it is a fundamental change to the model itself that should have a call out somewhere if not here.

File: integration_tests/dbt_project.yml
Comment (fivetran-sheringuyen) [inline integration_tests/dbt_project.yml:8]: Should the standard be to declare these identifiers globally or under `google_ads_source`?

File: models/google_ads.yml
Comment (fivetran-sheringuyen) [inline models/google_ads.yml:5]: is there a fundamental difference between UTMs vs URLs?

File: models/google_ads.yml
Comment (fivetran-sheringuyen) [inline models/google_ads.yml:191]: Is this a unique identifier for keyword_text? e.g. for ad_group level unique combos, we could use ad_group_id in place of ad_group_name, would this be the same logic?

File: models/google_ads__account_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__account_report.sql:6]: style-wise, I think these CTEs should look like (and similar in other models):

```
with stats as (

    select *
    from {{ var('account_stats') }}

), 

accounts as (
```

File: models/google_ads__ad_group_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__ad_group_report.sql:6]: Update styling

File: models/google_ads__ad_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__ad_report.sql:1]: Update styling

File: models/google_ads__campaign_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__campaign_report.sql:6]: Update styling

File: models/google_ads__keyword_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__keyword_report.sql:6]: Update styling

File: models/google_ads__utm_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__utm_report.sql:6]: Update styling

File: models/google_ads__utm_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__utm_report.sql:83]: I have a few questions here:
1. Would `ads.final_url` be different than `ads.source_final_urls`?

2. I noticed in FB that there could be URLs however the UTM fields could still be null. In which sense, I'd like to highlight my question above here as well
> Is there a fundamental difference between UTMs and URLs?

So if that is the case, should we be doing the non-null condition on UTMs extracted fields or URLs?

3. To make sure it filters for both the case of auto tagging enabled and not enabled, would it be more straight forward to do a `filtered_fields` CTE after this `fields` CTE for the below conditions instead?
`
    where utm_source is not null 
        or utm_medium is not null
        or utm_campaign is not null 
        or utm_content is not null
`

4. Additionally, is there a concept of `utm_term` for Google Ads? FB has 5 total.

**Review by fivetran-sheringuyen — COMMENTED** [review]
Hey @fivetran-joemarkiewicz !

Thanks for the awesome PR! This modeling package looks pretty awesome now. I had a couple in-line comments, let me know if you wanna chat!

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md:69]: Great call out. I just added a not to this section.

File: integration_tests/dbt_project.yml
Comment (fivetran-joemarkiewicz) [reply integration_tests/dbt_project.yml:8]: I feel like I have encountered situations where there are CircleCi issues and defining them globally has helped to reduce these errors. I think to avoid any possible errors it would be best to define globally.

File: models/google_ads__account_report.sql
Comment (fivetran-joemarkiewicz) [reply models/google_ads__account_report.sql:6]: Style updated across models.

File: models/google_ads__ad_group_report.sql
Comment (fivetran-joemarkiewicz) [reply models/google_ads__ad_group_report.sql:6]: Updated

File: models/google_ads__ad_report.sql
Comment (fivetran-joemarkiewicz) [reply models/google_ads__ad_report.sql:1]: Updated

File: models/google_ads__campaign_report.sql
Comment (fivetran-joemarkiewicz) [reply models/google_ads__campaign_report.sql:6]: Updated

File: models/google_ads__keyword_report.sql
Comment (fivetran-joemarkiewicz) [reply models/google_ads__keyword_report.sql:6]: Updated

File: models/google_ads__utm_report.sql
Comment (fivetran-joemarkiewicz) [reply models/google_ads__utm_report.sql:6]: Updated

File: models/google_ads__utm_report.sql
Comment (fivetran-joemarkiewicz) [reply models/google_ads__utm_report.sql:83]: > Would ads.final_url be different than ads.source_final_urls?

- `final_url` is the modified version of `final_urls`, whereas `source_final_urls` is the original version of `final_urls`. I felt that including the filter on the original field vs. the field we applied the concatenation would be better.

> Questions 2 and 3

- I think we should think more about the fundamental difference between UTMs and URLs. If there is a fundamental difference then I think we should go the route of filtering for only UTM populated fields. However, if there isn't then I think we should keep this filter as is and rename the model to be `google_ads__url_report`. I am becoming more keen to this idea. Especially as our original model for the ad reporting model was built for `url ad adapter` models. 🤔 Let's chat more about this one.

> Additionally, is there a concept of utm_term for Google Ads? FB has 5 total.
 
- Yes, Google has this concept. However, the auto tagging does't use this.

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md:91]: Updated

File: models/google_ads.yml
Comment (fivetran-joemarkiewicz) [reply models/google_ads.yml:191]: Not sure I am following this one 🤔

Comment (fivetran-sheringuyen) [reply models/google_ads.yml:191]: Your explanation makes sense! Resolving

Comment (fivetran-joemarkiewicz) [reply models/google_ads.yml:191]: Based off our slack convo I think we can keep it as it is.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks so much for this review @fivetran-sheringuyen!

I was able to address all your inline comments with the exception of the questions around the url vs utm debate. I think we should chat more about this live. Once we settle a firm approach on this I will regenerate the docs.

**Review by fivetran-sheringuyen — APPROVED** [review]

File: models/google_ads__url_report.sql
Comment (fivetran-sheringuyen) [inline models/google_ads__url_report.sql:63]: Hey Joe, I was just doing some work for Microsoft and was referencing this piece of code since both advertisers utilize the autotagging feature. 

Why are we doing a coalesce here rather than just using `google`, `cpc`, `campaigns.campaign_name`, `ad_groups.ad_group_name`? It looks like we're already using `dbt_utils.get_url_parameter('final_url', 'utm_source')` and etc in the `stg_*` models so this logic seems a little redundant. 

As for `utm_term` it looks like in Microsoft it's defined as the actual keyword terms from the keyword report. Not sure if that's different for Google.

**Comment by fivetran-joemarkiewicz** [comment]
Things to update before merging:
- Add not null tests

File: README.md
Comment (fivetran-reneeli) [inline README.md:69]: We should probably add language to include the end models as well. like "if you'd like to pass these additional metrics to the _end_ models, ..."

File: README.md
Comment (fivetran-reneeli) [inline README.md:70]: change `vetting` to `vetted`

**Pull request merged**
